internal/tracinginterceptor/interceptor.go

```diff
@@ -90,3 +148,63 @@ func (m *Interceptor) CallStream(ctx context.Context, req *transport.StreamReque
 	// TODO: implement
 	panic("implement me")
 }
+
+func updateSpanWithError(span opentracing.Span, err error) error {
+	if err == nil {
+		return err
+	}
+
+	ext.Error.Set(span, true)
+	if status := yarpcerrors.FromError(err); status != nil {
+		errCode := status.Code()
+		span.SetTag("rpc.yarpc.status_code", errCode.String())
+		span.SetTag("error.type", errCode.String())
+		return err
+	}
+
+	span.SetTag("error.type", "unknown_internal_yarpc")
+	return err
+}
+
+func updateSpanWithResponseError(span opentracing.Span, res *transport.Response, err error) error {
+	if err == nil && (res == nil || !res.ApplicationError) {
+		return err
+	}
+
+	ext.Error.Set(span, true)
+	status := yarpcerrors.FromError(err)
+	if status != nil {
+		errCode := status.Code()
+		span.SetTag("rpc.yarpc.status_code", errCode.String())
+		span.SetTag("error.type", errCode.String())
+		return err
+	}
+
+	if res != nil && res.ApplicationError {
+		span.SetTag("error.type", "application_error")
+		return err
+	}
+
```

Comment:
Could we also check the `ApplicationErrorMeta` as well to add more error details?

Reply:
Just as a reminder, we may use name and code, but should not add details.